esmvaltool/utils/cmorizers/obs/cmorize_obs_ESACCI-OC.py

```diff
@@ -0,0 +1,192 @@
+   Put all files under a single directory (no subdirectories with years)
+   in ${RAWOBS}/Tier2/ESACCI-OC
+
+Modification history
+   20190227-A_lova_to: written.
+
+"""
+
+import logging
+import os
+import glob
+import xarray as xr
+
+import iris
+import numpy as np
+
+from .utilities import (_set_global_atts, _fix_coords, _fix_var_metadata,
+                        _read_cmor_config, _save_variable, constant_metadata)
+
+logger = logging.getLogger(__name__)
+
+# read in CMOR configuration
+CFG = _read_cmor_config('ESACCI-OC.yml')
+
+
+def _fix_data(cube, var):
+    """Specific data fixes for different variables."""
+    logger.info("Fixing data ...")
+    with constant_metadata(cube):
+        if var == 'chl':
+            cube *= 1.e-06
+    return cube
+
+
+def _fix_attr_cmip5(out_dir, var):
+    """Adjust for CMIP5 standard"""
+    logger.info("Fix to CMIP5 standard...")
+    in_file = glob.glob(out_dir + '/OBS*' + var + '*.nc')[0]
+    DS = xr.open_dataset(in_file)
+    DS[var].attrs['coordinates'] = 'depth'
+    datt = {
+        'standard_name': 'depth',
+        'long_name': 'depth',
+        'units': 'm',
+        'axis': 'Z',
+        'positive': 'down',
+        '_FillValue': False
+    }
+
+    DS['depth'] = xr.DataArray(1., name='depth', attrs=datt)
+    DS.close()
+    DS.to_netcdf(in_file, mode='a')
+    return
+
+
+def extract_variable(var_info, raw_info, out_dir, attrs):
+    """Extract to all vars."""
+    var = var_info.short_name
+    cubes = iris.load(raw_info['file'])
+    rawvar = raw_info['name']
+
+    for cube in cubes:
+        if cube.var_name == rawvar:
+            _fix_var_metadata(cube, var_info)
+            _fix_coords(cube)
+            _fix_data(cube, var)
+            _set_global_atts(cube, attrs)
+            _save_variable(
+                cube,
+                var,
+                out_dir,
+                attrs,
+                local_keys=['coordinates'],
+                unlimited_dimensions=['time'],
+            )
+            _fix_attr_cmip5(out_dir, var)
+
+
+def da_coarsen(da, bin):
+    """ Rebin DataArray with user specified value"""
+    data = np.ma.masked_invalid(da.values)
+    lat = da.coords['lat'].values
+    lon = da.coords['lon'].values
+    # rebin
+    dd = data.shape
+    newda = da[:, :dd[1] // bin, :dd[2] // bin]
+    newda.values = data.reshape([dd[0], dd[1] // bin, bin, dd[2] // bin,
+                                 bin]).mean(-1).mean(-2)
+    newda.coords['lat'].values = lat.reshape([dd[1] // bin, bin]).mean(-1)
+    newda.coords['lon'].values = lon.reshape([dd[2] // bin, bin]).mean(-1)
+
+    return newda
+
+
+def merge_data(in_dir, out_dir, raw_info, bin):
+    """Merge all data into a single (regridded) file"""
+
+    var = raw_info['name']
+    file = raw_info['file']
+    do_bin = True if (bin % 2 == 0) & (bin != 0) else False
+    comment = ''
+    merged_file = os.path.join(out_dir, file + '_merged.nc')
+    #TODO remove 1997* here below before final publication
```

Comment:
I would remove it now, CMORizers should be tested in their full version, no matter how long it takes.

Reply:
@mattiarighi this is more a reminder for myself and I will remove it once that codacy and cmorizer implementation are finalized to make a complete test.

Reply:
I removed the TODO comment and *1997 in commit 1e7d257